### ISSUE TITLE
Text Layout issue, missing last letter

### ISSUE THREAD

**Issue by dbriard** [AvaloniaUI/Avalonia#13298]
With the following configuration, the x of CheckBox is missing.

```
<CheckBox UseLayoutRounding="False" 
		  FontFamily="Inter" 
		  FontSize="13">CheckBox</CheckBox>
```

It must be a precision issue as if I change one of the 3 properties below, the x appear correctly.

![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/f7c7ba65-5e3b-4a46-81ef-6c2a93f1807b)

**Comment by robloo** [comment]
Are you using any display scaling on your OS? I'm fairly certain these types of issues are tied to 125%+ display scaling on Windows.

**Comment by dbriard** [comment]
> Are you using any display scaling on your OS? I'm fairly certain these types of issues are tied to 125%+ display scaling on Windows.

I am using 250% but I have the exact same issue with 100%. You cannot reproduce?

**Comment by timunie** [comment]
Is it the FluentTheme? Which Font? Do you use the inter font? 

Side node: today I guess a font measure fix was merged. Can you probably test your case against nightly? Not sure if it is related to this issue. See: https://github.com/AvaloniaUI/Avalonia/pull/13268

**Comment by dbriard** [comment]
> Is it the FluentTheme? Which Font? Do you use the inter font?
> 
> Side node: today I guess a font measure fix was merged. Can you probably test your case against nightly? Not sure if it is related to this issue. See: #13268

The settings in the capture: inter 13, with no layout rounding. I just don't remember which Inter font, the one from Avalonia or one installed on my system as they can be a little different.

**Comment by Gillibald** [comment]
The font should not matter. This is some measure / arrange issue with unrounded text measurement values.

**Comment by dbriard** [comment]
> The font should not matter. This is some measure / arrange issue with unrounded text measurement values.

Sure it is just for you to reproduce the issue, as on a full page of controls, this is the only one that is missing a char, so not an obvious bug.

**Comment by timunie** [comment]
yeah, a minimal sample to reproduce would be ❤️ . Seems like a hard to catch issue though. I guess it must be some rounding issue somewhere.

**Comment by Developer-Alexander** [comment]
I don‘t know if this helps: I can see this phenomenon sometimes in an autosize column of a Grid.

**Comment by timunie** [comment]
@Developer-Alexander yeah then it could be related to grid layout, not text layout 🤔 . Maybe Auto should always round up.

**Comment by Developer-Alexander** [comment]
With this demo app I was able to catch the issue for a Checkbox with `UseLayoutRound="False"`. 
However I was not able to reproduce it for an autosize column of a grid.

![grafik](https://github.com/AvaloniaUI/Avalonia/assets/74363317/44118299-c495-42c6-a8e1-cbffe1f91830)


[MissingTrailingCharacter.zip](https://github.com/AvaloniaUI/Avalonia/files/13053853/MissingTrailingCharacter.zip)

**Comment by robloo** [comment]
Note that this is again very similar to a reported issue in the previews:

 * https://github.com/AvaloniaUI/Avalonia/issues/7502#issuecomment-1111638593
 * https://github.com/AvaloniaUI/Avalonia/issues/7502#issuecomment-1112568550

It was briefly fixed and then came back. Most recently it was mentioned here: 
 * https://github.com/AvaloniaUI/Avalonia/issues/13166#issuecomment-1752192701

If we have a reliable example that everyone can see now that's great!

> @Developer-Alexander yeah then it could be related to grid layout, not text layout 🤔 . Maybe Auto should always round up.

If it's the same issue I've been seeing in ToolTip it doesn't appear related to Grid. The ToolTip doesn't seem to have a Grid in the visual tree when displaying text.

**Comment by Gillibald** [comment]
I can't reproduce the issue with the sample you posted above

**Comment by Developer-Alexander** [comment]
> I can't reproduce the issue with the sample you posted above

A pity! I can reproduce it with this example at both 100% scaling and 175%.

**Comment by dbriard** [comment]
> I can't reproduce the issue with the sample you posted above

Here is a repro sample with the font embedded in resources as it does not happen with all versions of the Inter font. Please tell me if you can reproduce with this repro. Thanks a lot!
[CheckBoxBug.zip](https://github.com/AvaloniaUI/Avalonia/files/13058659/CheckBoxBug.zip)

**Comment by timunie** [comment]
@dbriard I can reproduce using your sample

**Comment by timunie** [comment]
Works if I remove the Padding btw.
